savant/utils/image.py

```diff
@@ -0,0 +1,528 @@
+        """Concatenates images along axis.
+
+        :param image: image to concatenate
+        :param axis: axis to concatenate. 0 - is vertical, 1 - is horizontal
+        :return: concatenated image
+        """
+        assert self._np_image.dtype == image.np_array.dtype, (
+            f'Images have different types {self._np_image.dtype } != '
+            f'{image.np_array.dtype}'
+        )
+        if axis == 0:
+            assert (
+                self.width == image.width
+            ), f'Images have different height {self.width} != {image.height}'
+
+            return CPUImage(
+                image=np.concatenate([self._np_image, image.np_array], axis=0)
+            )
+        elif axis == 1:
+            assert (
+                self.height == image.height
+            ), f'Images have different width {self.height} != {image.height}'
+            return CPUImage(
+                image=np.concatenate([self._np_image, image.np_array], axis=1)
+            )
+        else:
+            raise ValueError(f'Unknown axis {axis}')
+
+    def rotate(
+        self, angle: float, bbox: Optional[RBBox] = None
+    ) -> Tuple['CPUImage', RBBox]:
+        """Rotates image on angle.
+
+        :param angle: angle to rotate in degrees
+        :param bbox: image will be rotated on bbox.angle if bbox is not None
+        :return: rotated image
+        """
+
+        if bbox is not None:
+            rotation_matrix, resolution = get_rotation_matrix(
+                self, angle, (bbox.x_center, bbox.y_center)
+            )
+        else:
+            rotation_matrix, resolution = get_rotation_matrix(
+                self, angle, (self.width // 2, self.height // 2)
+            )
+        res = cv2.warpAffine(src=self._np_image, M=rotation_matrix, dsize=resolution)
+
+        if bbox is not None:
+            return CPUImage(res), RBBox(
+                x_center=resolution[0] / 2,
+                y_center=resolution[1] / 2,
+                width=bbox.width,
+                height=bbox.height,
+                angle=bbox.angle - angle,
+            )
+        return CPUImage(image=res), RBBox(
+            x_center=resolution[0] / 2,
+            y_center=resolution[1] / 2,
+            width=self.width,
+            height=self.height,
+            angle=angle,
+        )
+
+    def resize(
+        self,
+        resolution: Tuple[int, int],
+        method: str = 'fit',
+        interpolation: int = cv2.INTER_LINEAR,
+    ) -> 'CPUImage':
+        """Resizes image to resolution.
+
+        :param resolution: resolution to resize [width, height]
+        :param method: method to resize one of ['fit', 'scale']
+        :param interpolation: interpolation method.
+            Oone of [cv2.INTER_NEAREST, cv2.INTER_AREA, cv2.INTER_LINEAR,
+            cv2.INTER_CUBIC, cv2.INTER_LANCZOS4]
+        :return: resized image
+        """
+        new_resolution = tuple(map(int, resolution))
```

Comment:
`resolution` should already be a tuple of ints

Reply:
fixed